src/opm/parser/eclipse/EclipseState/Aquancon.cpp

```diff
@@ -18,107 +18,98 @@
  */
 #include <opm/parser/eclipse/EclipseState/Grid/FaceDir.hpp>
 #include <opm/parser/eclipse/EclipseState/Aquancon.hpp>
+
+#include <unordered_map>
 #include <utility>
 #include <algorithm>
 #include <iterator>
 #include <iostream>
 
 namespace Opm {
+
     namespace{
-        struct AquanconRecord{
-            std::vector<size_t> global_index_per_record;
 
-            // Variables constants
-            std::vector<std::shared_ptr<double>>  influx_coeff_per_record;  //Aquifer influx coefficient
-            std::vector<double>  influx_mult_per_record;   //Aquifer influx coefficient Multiplier
-            // Cell face to connect aquifer to
-            std::vector<int> face_per_record;
-            std::vector<int> record_index_per_record;
-        };
+        void add_cell(std::unordered_map<std::size_t, Aquancon::AquancCell>& work, Aquancon::AquancCell cell) {
+            auto cell_iter = work.find(cell.global_index);
+            if (cell_iter == work.end())
+                work.insert( std::make_pair(cell.global_index, std::move(cell)));
+            else {
+                /*
+                 */
+                auto& prev_cell = cell_iter->second;
+                if (prev_cell.aquiferID == cell.aquiferID) {
+                    if (prev_cell.influx_coeff.first != cell.influx_coeff.first)
+                        throw std::invalid_argument("Can not combine defaulted and not defaulted influx coefficient");
+
+                    if (prev_cell.influx_coeff.first) {
+                        if (cell.influx_coeff.second == 0)
+                            prev_cell.influx_coeff.second = 0;
+                        else
+                            prev_cell.influx_coeff.second += cell.influx_coeff.second;
+                    }
+                }
+            }
+        }
     }
 
+
     Aquancon::Aquancon(const EclipseGrid& grid, const Deck& deck)
     {
         if (!deck.hasKeyword("AQUANCON"))
             return;
 
-        std::vector<Opm::AquanconRecord> aqurecords;
-        // Aquifer ID per record
-        std::vector<int> aquiferID_per_record;
-        int m_maxAquID = 0;
-
+        std::unordered_map<std::size_t, Aquancon::AquancCell> work;
         const auto& aquanconKeyword = deck.getKeyword("AQUANCON");
```

Comment:
No - in that case it will only contain the last `AQUANCON` keyword. I guess that is an API design error in the `Deck` class . The call
```
auto deck_keyword = deck.getKeyword( "AQUANCON" );
```
will only return the *last* `AQUANCON` keyword. To get the them all you must do:
```
auto num_aquancon = deck.count("AQUANCON");
for (std::size_t iaq = 0; iaq < num_aquancon; iaqg++)
      const auto& kw = deck.getKeyword("AQUANCON", iaq);
```

I have updated the code to be as you indicate - I agree that seems most correct.